File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
num_dic = {"zero":"0", "one":"1", "two":"2", "three":"3", "four":"4", "five":"5", "six":"6", "seven":"7", "eight":"8", "nine":"9"}

def solution(s):
    answer = s
    for key, value in num_dic.items():
        answer = answer.replace(key, value)
    return int(answer)




def my_solution(s):
    dic = {'zero':'0', 'one':'1', 'two':'2', 'three':'3', 'four':'4', 'five':'5', 'six':'6', 'seven':'7', 'eight':'8', 'nine':'9'}
    
    answer = ''
    for i in range(len(s)):
        if s[i].isdigit():
            answer += s[i]
            i += 1
        elif s[i].isalpha():
            for length in range(3, 6):  # 영어 단어의 길이는 3에서 5까지의 범위로 가정
                word = s[i:i+length]
                if word in dic:
                    answer += dic[word]
                    i += length
                    break
    return int(answer)

def get_value_by_key(d, key):
    if key in d:
        return d[key]
    else:
        return None
    

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.*;

class Solution {
    public int solution(String s) {
                
        String[] temp_str = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        HashMap<String, String> num_dict = new HashMap<>();
        for (int i = 0; i < temp_str.length; i++) {
            num_dict.put(temp_str[i], Integer.toString(i));
        }
        
        // 가변 문자열 = StringBuilder 사용
        StringBuilder temp = new StringBuilder();
        StringBuilder answer = new StringBuilder();
        
        for (char c : s.toCharArray()) {
            if (Character.isLetter(c)) {
                temp.append(c);
                if (num_dict.containsKey(temp.toString())) {
                    answer.append(num_dict.get(temp.toString()));
                    temp.setLength(0); // 초기화
                }
            }
            else {
                answer.append(c);
                temp.setLength(0); // 초기화 
            }
        }
        return Integer.parseInt(answer.toString());
    }
    
    static void print(Object o) {  
        System.out.println(o);    
    }
}
